src/NuGet.Services.Staging.Search/DatabaseSearchResultsFormatter.cs

```diff
@@ -0,0 +1,102 @@
+        {
+            if (pathCalculator == null)
+            {
+                throw new ArgumentNullException(nameof(pathCalculator));
+            }
+
+            _pathCalculator = pathCalculator;
+        }
+
+        public JObject FormatSearchResults(IEnumerable<PackageMetadata> packagesMetadata)
+        {
+            // If there are several packages with the same Id, but different version,
+            // we would return one result per id
+            var idGroups = packagesMetadata.GroupBy(x => x.Id);
+            var formatedPackages = idGroups.Select(FormatPackageResult);
+
+            return new JObject
+            {
+                {
+                    "@context", new JObject
+                    {
+                        {
+                            "@vocab", "http://schema.nuget.org/schema#"
+                        }
+                    }
+                },
+                {
+                    "data", new JArray(formatedPackages)
+                }
+            };
+        }
+
+        private JObject FormatPackageResult(IEnumerable<PackageMetadata> packagesMetadata)
+        {
+            // Sort packages from 
+            var sortedPackages = packagesMetadata.OrderBy(x => x, new VersionComparer());
+            var newestVersion = sortedPackages.Last();
+
+            var authorsList = newestVersion.Authors.Split(',').Select(s => s.Trim());
+            var tagsList = newestVersion.Tags.Split(' ');
+
+            var versionsList = sortedPackages.Select(FormatVersionResult);
+
+            return new JObject
+            {
+                { "@id", _pathCalculator.GetPackageVersionRegistrationAddress(newestVersion.Id, newestVersion.Version).ToString() },
+                { "@type", "Package" },
+                { "authors", new JArray(authorsList) },
+                { "description",newestVersion.Description },
+                { "iconUrl", newestVersion.IconUrl },
+                { "id", newestVersion.Id},
+                { "licenseUrl", newestVersion.LicenseUrl },
+                { "projectUrl", newestVersion.ProjectUrl },
+                { "registration", _pathCalculator.GetPackageRegistrationIndexAddress(newestVersion.Id).ToString() },
+                { "tags", new JArray(tagsList) },
+                { "title", newestVersion.Title },
+                { "totalDownloads", 0 },
+                { "version", newestVersion.Version },
+                { "versions", new JArray(versionsList) }
+            };
+        }
+
+        private JObject FormatVersionResult(PackageMetadata packageMetadata)
+        {
+            return new JObject
+            {
+                { "@id", _pathCalculator.GetPackageVersionRegistrationAddress(packageMetadata.Id, packageMetadata.Version).ToString() },
+                { "downloads", 0 },
+                { "version", packageMetadata.Version }
+            };
+        }
+
+        private class VersionComparer : IComparer<PackageMetadata>

```

Comment:
It's actually a PackageMetadataVersionComparer :-)
The base libraries do have a NuGetVersionComparer, can that one be used instead?